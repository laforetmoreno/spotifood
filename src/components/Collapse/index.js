/* eslint-disable react/jsx-props-no-spreading */
import React, { useState } from 'react';
import useCollapse from 'react-collapsed';
import { node } from 'prop-types';

import filter from 'assets/filter.svg';

import styles from './index.module.scss';

const Collapse = ({ children }) => {
  /* eslint-disable-next-line */
  const [renderChildren, setRenderChildren] = useState(false);
  const { getToggleProps, getCollapseProps, isExpanded } = useCollapse({
    defaultExpanded: true,
    onExpandStart() {
      setRenderChildren(true);
    },
    onCollapseEnd() {
      setRenderChildren(false);
    },
  });

  return (
    <>
      <button
        {...getToggleProps({ style: { display: 'block' } })}
        type="button"
        className={styles.btn}
      >
        <img className={isExpanded ? undefined : styles.down} alt="Filter" src={filter} />
      </button>
      <div {...getCollapseProps()}>{children}</div>
    </>
  );
};

Collapse.propTypes = {
  children: node,
};

Collapse.defaultProps = {
  children: <div />,
};

export default Collapse;
