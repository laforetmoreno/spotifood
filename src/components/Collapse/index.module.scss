.btn {
  background: transparent;
  border: none;
  width: 100%;
  text-align: right;
  outline: none;
  cursor: pointer;
  padding-top: 10px;

  img {
    height: 20px;
  }
}
