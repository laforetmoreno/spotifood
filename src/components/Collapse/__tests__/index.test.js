import React from 'react';
import renderer from 'react-test-renderer';
import Collapse from '../index';

describe('<Collapse />', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Collapse>
          <h1>test</h1>
        </Collapse>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
