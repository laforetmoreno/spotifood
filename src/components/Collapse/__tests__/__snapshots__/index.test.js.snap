// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Collapse /> renders correctly 1`] = `
Array [
  <div
    className="wrapper collapse"
  >
    <h1>
      test
    </h1>
  </div>,
  <button
    className="btn"
    onClick={[Function]}
    type="button"
  >
    <img
      alt="Arrow"
      src="arrow.svg"
    />
  </button>,
]
`;
